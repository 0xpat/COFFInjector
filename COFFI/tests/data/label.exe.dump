
label.exe:     format de fichier pei-i386
label.exe
architecture: i386, fanions 0x0000012f:
HAS_RELOC, EXEC_P, HAS_LINENO, HAS_DEBUG, HAS_LOCALS, D_PAGED
adresse de départ 0x00401000

Characteristics 0x102
	executable
	32 bit words

Time/Date		Thu Jan 01 01:00:00 1970
Magic			010b	(PE32)
MajorLinkerVersion	4
MinorLinkerVersion	0
SizeOfCode		00000200
SizeOfInitializedData	00000000
SizeOfUninitializedData	00000000
AddressOfEntryPoint	00001000
BaseOfCode		00001000
BaseOfData		00000000
ImageBase		00400000
SectionAlignment	00001000
FileAlignment		00000200
MajorOSystemVersion	4
MinorOSystemVersion	0
MajorImageVersion	0
MinorImageVersion	0
MajorSubsystemVersion	4
MinorSubsystemVersion	0
Win32Version		00000000
SizeOfImage		00004000
SizeOfHeaders		00000200
CheckSum		00000000
Subsystem		00000003	(Windows CUI)
DllCharacteristics	00000000
SizeOfStackReserve	00100000
SizeOfStackCommit	00001000
SizeOfHeapReserve	00100000
SizeOfHeapCommit	00001000
LoaderFlags		00000000
NumberOfRvaAndSizes	00000010

The Data Directory
Entry 0 00000000 00000000 Export Directory [.edata (or where ever we found it)]
Entry 1 00003000 0000005d Import Directory [parts of .idata]
Entry 2 00000000 00000000 Resource Directory [.rsrc]
Entry 3 00000000 00000000 Exception Directory [.pdata]
Entry 4 00000000 00000000 Security Directory
Entry 5 00000000 00000000 Base Relocation Directory [.reloc]
Entry 6 00000000 00000000 Debug Directory
Entry 7 00000000 00000000 Description Directory
Entry 8 00000000 00000000 Special Directory
Entry 9 00000000 00000000 Thread Storage Directory [.tls]
Entry a 00000000 00000000 Load Configuration Directory
Entry b 00000000 00000000 Bound Import Directory
Entry c 00000000 00000000 Import Address Table Directory
Entry d 00000000 00000000 Delay Import Directory
Entry e 00000000 00000000 CLR Runtime Header
Entry f 00000000 00000000 Reserved

There is an import table in .idata at 0x403000

The Import Tables (interpreted .idata section contents)
 vma:            Hint    Time      Forward  DLL       First
                 Table   Stamp     Chain    Name      Thunk
 00003000	00003028 00000000 00000000 00003052 00003034

	DLL Name: msvcrt.dll
	vma:  Hint/Ord Member-Name Bound-To
	3040	    0  exit
	3048	    0  printf

 00003014	00000000 00000000 00000000 00000000 00000000

Sections :
Idx Name          Taille    VMA       LMA       Off fich  Algn
  0 .text         00000033  00401000  00401000  00000200  2**2
                  CONTENTS, ALLOC, LOAD, READONLY, CODE, DATA
  1 .rdata        00000004  00402000  00402000  00000400  2**2
                  CONTENTS, ALLOC, LOAD, READONLY, DATA
  2 .idata        0000005d  00403000  00403000  00000600  2**2
                  CONTENTS, ALLOC, LOAD, DATA
SYMBOL TABLE:
aucun symbole


