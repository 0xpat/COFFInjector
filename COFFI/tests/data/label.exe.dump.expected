Dump of file data/label.exe

File Header
  Machine:                      014C
  Number of Sections:           0003

  TimeDateStamp:                00000000 (1970-01-01 01:00:00)
  PointerToSymbolTable:         00000000
  NumberOfSymbols:              00000000
  SizeOfOptionalHeader:         00E0
  Characteristics:              0102

Optional Header
  Magic                         010B
  linker version                4.0
  size of code                  0200
  size of initialized data      00000000
  size of uninitialized data    00000000
  entrypoint RVA                00001000
  base of code                  00001000
  base of data                  00000000

  image base                    0000000000400000
  section align                 00001000
  file align                    00000200
  required OS version           4.0
  image version                 0.0
  subsystem version             4.0
  Win32 Version                 0
  size of image                 00004000
  size of headers               00000200
  checksum                      00000000
  Subsystem                     0003
  DLL flags                     0000
  stack reserve size            0000000000100000
  stack commit size             0000000000001000
  heap reserve size             0000000000100000
  heap commit size              0000000000001000
  RVAs & sizes                  00000010

Data Directory
  EXPORT_TABLE            rva: 00000000  size: 00000000
  IMPORT_TABLE            rva: 00003000  size: 0000005D
  RESOURCE_TABLE          rva: 00000000  size: 00000000
  EXCEPTION_TABLE         rva: 00000000  size: 00000000
  CERTIFICATE_TABLE       rva: 00000000  size: 00000000
  BASE_RELOCATION_TABLE   rva: 00000000  size: 00000000
  DEBUG                   rva: 00000000  size: 00000000
  ARCHITECTURE            rva: 00000000  size: 00000000
  GLOBAL_PTR              rva: 00000000  size: 00000000
  TLS_TABLE               rva: 00000000  size: 00000000
  LOAD_CONFIG_TABLE       rva: 00000000  size: 00000000
  BOUND_IMPORT            rva: 00000000  size: 00000000
  IAT                     rva: 00000000  size: 00000000
  DELAY_IMPORT_DESCRIPTOR rva: 00000000  size: 00000000
  COMPLUS_RUNTIME_HEADER  rva: 00000000  size: 00000000
  RESERVED                rva: 00000000  size: 00000000

Section Table
  000 .text     VirtSize: 00000033  VirtAddr:  00001000
    raw data offs:   00000200  raw data size: 00000200
    relocation offs: 00000000  relocations:   00000000
    line # offs:     00000000  line #'s:      00000000
    characteristics: 60000060
      CODE INITIALIZED_DATA EXECUTE READ

  001 .rdata    VirtSize: 00000004  VirtAddr:  00002000
    raw data offs:   00000400  raw data size: 00000200
    relocation offs: 00000000  relocations:   00000000
    line # offs:     00000000  line #'s:      00000000
    characteristics: 40000040
      INITIALIZED_DATA READ

  002 .idata    VirtSize: 0000005D  VirtAddr:  00003000
    raw data offs:   00000600  raw data size: 00000200
    relocation offs: 00000000  relocations:   00000000
    line # offs:     00000000  line #'s:      00000000
    characteristics: C0000040
      INITIALIZED_DATA READ WRITE


